src/Collections/Collection.php

```diff
@@ -415,6 +425,14 @@ public function append(Model $model)
         return ($result === null) ? $model : $result;
     }
 
+    /**
+     * Removes all intersections
+     */
+    final public function clearIntersections()
+    {
+        $this->intersections = [];
+    }
```

Comment:
Would it make sense to return `$this` here to keep consistent with the behaviour of `clearGroups()`

Reply:
Good point, will change this.